### PR TITLE
feat(tiering): Simple snapshotting

### PR DESCRIPTION
**Pull request by dranikpg**
* Adds support for snapshotting for tiered entries

### CODE REVIEW

**Comment by dranikpg** [comment]
Ah dang, it conflicts with defragmentation 🤨

**Comment by dranikpg** [comment]
Now we don't fetch and don't offload while snapshotting is in progress. I alslo fixed the bug that occured when we did this

File: src/server/snapshot.cc [hunk 1]
```diff
@@ -282,12 +283,30 @@ void SliceSnapshot::SerializeEntry(DbIndex db_indx, const PrimeKey& pk, const Pr
...
+  } else {
+    io::Result<uint8_t> res = serializer->SaveEntry(pk, pv, expire_time, db_indx);
+    CHECK(res);
+    ++type_freq_map_[*res];
+  }
 }
 
 bool SliceSnapshot::PushSerializedToChannel(bool force) {
+  // Bucket serialization might have accumulated some delayed values.
+  // Becuause we can finally block in this function, we'll await and serialize them
```
Comment (romange) [inline src/server/snapshot.cc:303]: Becuause -> Because

File: src/server/snapshot.h [hunk 2]
```diff
@@ -141,6 +150,7 @@ class SliceSnapshot {
   DbIndex current_db_;
 
   std::unique_ptr<RdbSerializer> serializer_;
+  std::deque<DelayedEntry> delayed_entries_;  // collected during atomic bucket traversal
```
Comment (romange) [inline src/server/snapshot.h:153]: why deque? vector (stack) won't work here?

Comment (dranikpg) [reply src/server/snapshot.h:153]: Actually, it will

**Review by romange — APPROVED** [review]

**Pull request merged**